pkg/ring/replication_strategy.go

```diff
@@ -26,14 +27,15 @@ func (r *Ring) replicationStrategy(ingesters []IngesterDesc, op Operation) (
 	// Skip those that have not heartbeated in a while. NB these are still
 	// included in the calculation of minSuccess, so if too many failed ingesters
 	// will cause the whole write to fail.
-	liveIngesters = make([]IngesterDesc, 0, len(ingesters))
-	for _, ingester := range ingesters {
-		if r.IsHealthy(&ingester, op) {
-			liveIngesters = append(liveIngesters, ingester)
+	for i := 0; i < len(ingesters); {
+		if r.IsHealthy(&ingesters[i], op) {
+			i++
 		} else {
+			ingesters = append(ingesters[:i], ingesters[i+1:]...)
 			maxFailure--
 		}
 	}
+	liveIngesters = ingesters
```

Comment:
I would fully support the removal of the use of named return parameters; I find them hard to follow in the review...